vertexai/context-caching/contextcaching_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package contextcaching
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/vertexai/genai"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+// TestContextCaching tests createContextCache, getContextCache, useContextCache,
+// updateContextCache, deleteContextCache.
+func TestContextCaching(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	buf := new(bytes.Buffer)
```

Comment:
Why not just use stdout? It might be useful to see the output in the event that the test fails.

Reply:
In the tests we don't use `os.Stdout`, nor `t.Logf`.  We do use `t.Errorf` and `t.Fatalf` though.  See [CONTRIBUTING.md](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#print-to-an-iowriter-for-snippets).

Reply:
I would have thought that just setting `w` to `io.Stdout` would be fine for tests. I'm not sure I understand the reasoning that we should throw out this output in tests but I'm willing to defer.

Reply:
I guess writing to stdout/stderr/t.output  is not necessary for CI/CD, and may generate noise in the logs

Reply:
Go's testing module suppresses stdio output unless tests fail so it shouldn't generate noise. It's not that big a deal though. 🤷